docs/learn/smart-contract-developers/create-nft-collection-with-lsp7-tokenId.md

```diff
@@ -0,0 +1,953 @@
+        bytes memory lsp4MetadataURIOfLSP7_
+    ) public returns (address lsp7TokenAddress) {
+        // deploy the LSP7Token and set the address as tokenId
+        LSP7Token lsp7Token = new LSP7Token(
+            nameOfLSP7_,
+            symbolOfLSP7_,
+            address(this), // owner of the LSP7Token is this contract (LSP8Token contract)
+            receiverOfInitialTokens_,
+            lsp4TokenType_,
+            isNonDivisible_,
+            totalSupplyOfLSP7_,
+            lsp4MetadataURIOfLSP7_
+        );
+
+        lsp7TokenAddress = address(lsp7Token);
+
+        // convert the address of the LSP7Token to bytes32 to use it as tokenId
+        bytes32 tokenId = bytes32(uint256(uint160(lsp7TokenAddress)));
+
+        /*
+          owner of the tokenId is this contract
+          tokenId is the address of the newly deployed LSP7Token
+          force is true since here the owner of the tokenId is this contract
+          data is empty
+        */
+        _mint(address(this), tokenId, true, "");
+    }
+
+    // override the _setDataForTokenId function to set the data on the LSP7Token itself
+    function _setDataForTokenId(
+        bytes32 tokenId,
+        bytes32 dataKey,
+        bytes memory dataValue
+    ) internal override {
+        // setData on the LSP7Token
+        LSP7Token(payable(address(uint160(uint256(tokenId))))).setData(
+            dataKey,
+            dataValue
+        );
+
+        emit TokenIdDataChanged(tokenId, dataKey, dataValue);
+    }
+
+    // override the _getDataForTokenId function to get the data from the LSP7Token itself
+    function _getDataForTokenId(
+        bytes32 tokenId,
+        bytes32 dataKey
+    ) internal view override returns (bytes memory dataValues) {
+        return
+            LSP7Token(payable(address(uint160(uint256(tokenId))))).getData(
+                dataKey
+            );
+    }
+}
+```
+
+</details>
+
+## Scripts
+
+Now that we have the contracts ready to be deployed, let's create a script that will deploy the LSP8 token and one script that will mint the LSP7 tokens.
+
+:::note
+
+Please make sure you compile your newly created contracts before creating the scripts.
+You can run `npx hardhat compile` to compile your contracts.
+Since the contracts are quite big, you may run into a compilation error. If this is the case, you should update your compiler settings in the `hardhat.config.ts` file.
+This is what we used to compile the contracts:
+
+<details>
+<summary>Click to expand/collapse the script</summary>
+```typescript
+  solidity: {
+    version: "0.8.20",
+    settings: {
+      optimizer: {
+        enabled: true,
+        runs: 200,
+      },
+    },
+  }
+````
+
+</details>
+:::
+
+### Create the metadata JSONs files
+
+:::note
+In this part we will not cover the part where you first need to upload your JSONs files on IPFS and retrieve their CIDs.
+We have done it and will add them directly in the following scripts files.
```

Comment:
i ssuggest to put link and direct the user to the rights page so the user can learn about bhow to generate the files (erc725.js) or how to read them eth (link to lsp2 schema, and so on)

Reply:
I dont think we have one that is up to date. I am waiting for Andreas to release his lib so we can create a new one. I was hustling finding these info and those are not complete or updated

Reply:
ok good point

Reply:
Might want to make a note that IPFS folders work in this situation and the base URI can be a folder CID. I added a doc PR to the LIP that when constructing the URL we need to use encodeURI which will leave / intact. So deeper folder levels are also possible.

Reply:
NOTE: there are currently folder level IPFS links out there already. Here is an example 
https://bafybeif3wv62jah7jsrtb54yt6n5n3dg72d6mxgnadmpdtrcec2gjqb7wy.ipfs.4everland.io/